server/src/main/scala/org/http4s/server/middleware/CSRFMiddleware.scala

```diff
@@ -0,0 +1,196 @@
+package org.http4s.server.middleware
+
+import java.nio.charset.StandardCharsets
+import java.security.{MessageDigest, SecureRandom}
+import java.time.Clock
+import javax.crypto.{KeyGenerator, Mac, SecretKey}
+import javax.crypto.spec.SecretKeySpec
+import java.util.Base64
+
+import org.http4s.{MaybeResponse, Request, Response, Status, Cookie}
+import org.http4s.server.Middleware
+import org.http4s.headers.{Cookie => HCookie}
+import org.http4s.util.{CaseInsensitiveString, NonEmptyList}
+
+import scalaz.{Kleisli, OptionT}
+import scalaz.concurrent.Task
+import scalaz.syntax.std.boolean._
```

Comment:
Starting in 0.18, we've just done an alphabetical sort of the imports.  This isn't enforced on the legacy branches, but it'll fit in nicer if we start down here.